server/http_hooks.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"html/template"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+// ServeHTTP allows the plugin to implement the http.Handler interface. Requests destined for the
+// /plugins/{id} path will be routed to the plugin.
+func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
+	var action *Action
+	json.NewDecoder(r.Body).Decode(&action)
```

Comment:
Should handle the error, or is it fine to just use the `action == nil` check below as a proxy for the error?

Reply:
It's a nit, but 1/5 we should be displaying the errors